src/Orleans/Runtime/IGrainLifeCycle.cs

```diff
@@ -0,0 +1,26 @@
+﻿
+namespace Orleans.Runtime
+{
+    /// <summary>
+    /// Stages of a grains lifecycle.
+    /// TODO: Add more later, see ActivationInitializationStage
+    /// Full grain lifecycle, including register, state setup, and 
+    ///   stream cleanup should all eventually be triggerd by the 
+    ///   grian lifecycle.
+    /// </summary>
+    public enum GrainLifecyleStage
+    {
+        //None,
+        //Register,
+        SetupState,  // Setup grain state prior to activation
+        //InvokeActivate,
+        //Completed
```

Comment:
Probably these well-known stages should have an explicit value, and leave values in between so that users can provide a value in between